### PR TITLE
Do not build static libraries

### PR DESCRIPTION
**Pull request by hmaarrfk**
Is there an opposition to disabling the static libraries? I notice that this has been being built since 2018.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
We should be able to add teh downstreams again: https://github.com/conda-forge/gnutls-feedstock/pull/24

**Comment by hmaarrfk** [comment]
I'm going to let this sit for a few days.

Hopefully somebody can review.

We will need to migrate gnutls to the new pin to enable the downstream tests again.

**Comment by hmaarrfk** [comment]
https://github.com/conda-forge/gnutls-feedstock/pull/24

**Comment by erykoff** [comment]
I’m curious but what’s the reason for disabling the static libraries?

**Comment by hmaarrfk** [comment]
They mostly take up needless space on people's computers (well mine really!).

See https://github.com/conda-forge/cfep/blob/main/cfep-18.md for a more thorough explination.

I'm affraid to merge in case you are relying on this for something particular

**Comment by erykoff** [comment]
This is just needed for gnutls, which is needed for emacs.  If gnutls works without the static libraries, then I'm fine with the change.

**Comment by hmaarrfk** [comment]
are you able to help with https://github.com/conda-forge/gnutls-feedstock/pull/24 by any chance?

**Comment by hmaarrfk** [comment]
ok i've started a compilation manually removing the .a files before the mkae command starts. it should pick it up from the .so for sure now.
https://github.com/conda-forge/gnutls-feedstock/pull/25

**Comment by hmaarrfk** [comment]
I looked through the GNUTLS logs of the last build on the master branch and they have the lines:
```
2020-11-20T19:05:14.3769381Z    INFO (gnutls,lib/libgnutls.so.30.27.0): Needed DSO lib/libnettle.so.8 found in conda-forge::nettle-3.6-he412f7d_0
2020-11-20T19:05:14.3771050Z    INFO (gnutls,lib/libgnutls.so.30.27.0): Needed DSO lib/libhogweed.so.6 found in conda-forge::nettle-3.6-he412f7d_0
```
which means they are in fact using the dynamic libraries.

I also made a proof of concept PR that gnutls bulids fine without the static libraries and they work just fine.

Let me know if you need anything more.
https://github.com/conda-forge/gnutls-feedstock/pull/25

**Comment by erykoff** [comment]
Sorry I haven't gotten back to this.  It all seems like a good idea but I'm worried that https://github.com/conda-forge/gnutls-feedstock/pull/25 isn't compiling cleanly on osx-64.  Is that related to these changes?

**Comment by hmaarrfk** [comment]
I believe that OSX is related to other issues.

There is a note of an expected bug in a particular version of OSX.

Finally, i believe it is compiling, but not passing tests.
The same failure should be observable in the build with the static libraries

**Comment by hmaarrfk** [comment]
no issue on the delay

**Comment by sachalau** [comment]
Sorry for the delay as well ! Shall I try and update to the latest nettle version as well (3.8 just dropped) or do you think we shall merge sequentially ?

**Comment by hmaarrfk** [comment]
@sachalau it would be good to syncrhonize this with a migrator, that way people will know to fix their recipes.

**Comment by sachalau** [comment]
Oh no I think this went straight way above my competencies :( Somehow I was the first author of the nettle/gnutls recipes but I lack the skills to maintain them ! 

By using a migrator, do you mean using this exact PR https://github.com/conda-forge/nettle-feedstock/pull/12 or can I prepare another PR with the version update + disabling of static libraries ? 

I've seen you've closed the automatic PR for the gnutls pin as well. Not sure if related ?

**Comment by hmaarrfk** [comment]
https://github.com/conda-forge/nettle-feedstock/pull/12 is good practice. you likely can add the small modifications from here there. Try to fix it up, then ping me if you get stuck and can't understand the logs.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,10 +30,10 @@ test:
...
     - test -f ${PREFIX}/lib/libnettle${SHLIB_EXT}
     - test -f ${PREFIX}/lib/libhogweed${SHLIB_EXT}
-    - test -f ${PREFIX}/lib/libnettle.a
-    - test -f ${PREFIX}/lib/libhogweed.a
-  downstreams:
-    - gnutls
+    - test ! -f ${PREFIX}/lib/libnettle.a
+    - test ! -f ${PREFIX}/lib/libhogweed.a
+  # downstreams:
+  #   - gnutls
```
Comment (hmaarrfk) [inline recipe/meta.yaml:36]: you might want to comment these lines too. when you update to 3.8

**Pull request closed without merging**